agent/acs/handler/payload_handler_test.go

```diff
@@ -907,3 +907,50 @@ func TestHandleUnrecognizedTask(t *testing.T) {
 	tester.payloadHandler.handleUnrecognizedTask(ecsacsTask, errors.New("test error"), payloadMessage)
 	wait.Wait()
 }
+
+func TestPayloadHandlerAddedLogRouterData(t *testing.T) {
+	tester := setup(t)
+	defer tester.ctrl.Finish()
+
+	var addedTask *apitask.Task
+	tester.mockTaskEngine.EXPECT().AddTask(gomock.Any()).Do(
+		func(task *apitask.Task) {
+			addedTask = task
+		})
+
+	payloadMessage := &ecsacs.PayloadMessage{
+		Tasks: []*ecsacs.Task{
+			{
+				Arn: aws.String("arn"),
+				Containers: []*ecsacs.Container{
+					{
+						LogRouter: &ecsacs.LogRouter {
+							Config: &ecsacs.LogRouterConfig{
+								Type: aws.String("arn"),
+								Value :  aws.String("asm-arn"),
+							},
+							EnableECSLogMetaData:aws.Bool(true),
+							Type: aws.String("fluentd"),
```

Comment:
nits: The spacing here is a little bit weird, and I'd do `s3-arn` (or a valid looking arn), instead of `asm-arn`.

Reply:
Fixed